### PR TITLE
Migrate EfficentNetV2 

### PR DESCRIPTION
**Pull request by soma2000-lang**
Continued from  #1645 which accidentally got closed  due to force push
Solves #1626 

## Who can review?
@jbischof @ianstenbit @haifeng-jin @LukeWood

### CODE REVIEW

**Comment by soma2000-lang** [comment]
All  the lint tests of all my prs are failing due to this .Currently investigating why.
image

**Comment by jbischof** [comment]
@soma2000-lang you just need to run `./shell/format.sh`

**Comment by ID6109** [comment]
Actually, this is likely due to a messy merge because line length issues are clearly extending beyond the files this PR intends to change. Refer to [this](https://github.com/keras-team/keras-cv/pull/1552).

**Comment by soma2000-lang** [comment]
@jbischof I have runned that.still the same the lint is failing

**Comment by soma2000-lang** [comment]
@jbischof @ID6109 The problem  is that I have not made changes  to those files (whose Lint is failing).I have used # noqa at 
places in my files and have runned shell/format.sh

File: keras_cv/models/backbones/efficientnetv2/efficientnet_v2_backbone.py [hunk 1]
```diff
@@ -592,42 +572,15 @@ class EfficientNetV2(keras.Model):
...
-      ValueError: if `classifier_activation` is not `"softmax"` or `None` when
-        using a pretrained top layer.
     """
 
     def __init__(
         self,
+        *,
         include_rescaling,
-        include_top,
         width_coefficient,
```
Comment (LukeWood) [inline keras_cv/models/backbones/efficientnetv2/efficientnet_v2_backbone.py:584]: @jbischof what do you want to do about things like "width_coefficient" and "depth_coefficient" that are generally only used by the wrapping classes?  We typically dont expect users to configure them; but I guess we just document them and leave them in?

File: keras_cv/models/backbones/efficientnetv2/efficientnet_v2_backbone.py [hunk 2]
```diff
@@ -831,246 +739,257 @@ def __init__(
...
-        input_shape=input_shape,
-        input_tensor=input_tensor,
-        pooling=pooling,
-        num_classes=num_classes,
-        classifier_activation=classifier_activation,
+
+class EfficientNetV2SBackbone(EfficientNetV2Backbone):
+    def __new__(
+        cls,
+        include_rescaling=True,
```
Comment (LukeWood) [inline keras_cv/models/backbones/efficientnetv2/efficientnet_v2_backbone.py:774]: these subclasses should be passing the __init__ constructor of `super()` values for width_coefficient and and depth_coefficient.

See line 912 in the current file:

```

    return EfficientNetV2(
        include_rescaling=include_rescaling,
        include_top=include_top,
        width_coefficient=1.0,
        depth_coefficient=1.1,
        default_size=240,
        model_name="efficientnetv2-b1",
        weights=parse_weights(weights, include_top, "efficientnetv2b1"),
        input_shape=input_shape,
        input_tensor=input_tensor,
        pooling=pooling,
        num_classes=num_classes,
        classifier_activation=classifier_activation,
```

You'll want to preserve this!

**Review by LukeWood — CHANGES_REQUESTED** [review]
overall this looks quite good - there are just a few changes that need to be made.  Thanks @soma2000-lang !

**Comment by soma2000-lang** [comment]
@jbischof @ID6109 The problem  is that I have not made changes  to those files (whose Lint is failing).I have used # noqa at 
places in my files and have runned shell/format.sh

**Comment by soma2000-lang** [comment]
Hi @cosmo3769 you can collaborate but just a gentle request ,please dont open PRs for issues for which you can see pr already  is opened.

**Pull request closed without merging**